docs/src/userguide/wrapping_CPlusPlus.rst

```diff
@@ -187,34 +79,47 @@ class name from Rectangle.h and adjust for Cython syntax, so now it becomes::
 Add public attributes
 ^^^^^^^^^^^^^^^^^^^^^^
 
-We now need to declare the attributes and methods for use on Cython::
+We now need to declare the attributes and methods for use on Cython. We put those declarations
+in a file called :file:`Rectangle.pxd`. You can see it as a header file
+which is readable by Cython:
 
-    cdef extern from "Rectangle.h" namespace "shapes":
-        cdef cppclass Rectangle:
-            Rectangle() except +
-            Rectangle(int, int, int, int) except +
-            int x0, y0, x1, y1
-            int getArea()
-            void getSize(int* width, int* height)
-            void move(int, int)
-
-Note that the constructor is declared as "except +".  If the C++ code or
+.. literalinclude:: ../../examples/userguide/wrapping_CPlusPlus/Rectangle.pxd
+
+Note that the constructor is declared as "except +". If the C++ code or
 the initial memory allocation raises an exception due to a failure, this
 will let Cython safely raise an appropriate Python exception instead
-(see below).  Without this declaration, C++ exceptions originating from
+(see below). Without this declaration, C++ exceptions originating from
```

Comment:
I personally find double spaces between sentences helpful as they provide a clearer separation. Also, at least Emacs can make use of them for jumping from sentence to sentence. I'm not suggesting to blindly add them everywhere, but I also don't see a reason to change them to single spaces when they are there already.